js/bitmart.js

```diff
@@ -669,7 +769,9 @@ module.exports = class bitmart extends Exchange {
          */
         const spot = await this.fetchSpotMarkets (params);
         const contract = await this.fetchContractMarkets (params);
-        return this.arrayConcat (spot, contract);
+        const margin = await this.fetchMarginMarkets (params);
+        const spotContract = this.arrayConcat (spot, contract);
+        return this.arrayConcat (spotContract, margin);
```

Comment:
I'm afraid, this will overwrite the market ids and symbols indexes, if margin pairs have the same symbols as spot pairs.

Reply:
> I'm afraid, this will overwrite the market ids and symbols indexes, if margin pairs have the same symbols as spot pairs.

What would be the best approach here? Should I just add the margin endpoint to fetchSpotMarkets and set margin to true if the symbol is defined and leave it at that?